benchmark_custom.py

```
@@ -0,0 +1,49 @@
+        root_hex = compute_merkle_root(file_path, chunk_size=chunk_size)
+        end_time = time.perf_counter()
+
+        root_time = end_time - start_time
+        print(f"compute_merkle_root ({size_mb:.2f} MB file): {root_time:.4f} seconds")
+        print(f"Merkle Root: {root_hex}")
+
+        # Benchmark generate_merkle_proof
+        start_time = time.perf_counter()
+        # Generate proof for a chunk (e.g. chunk 10 if there are enough, otherwise chunk 0)
+        chunk_index = 10 if size_mb > 10 else 0
+        proof = generate_merkle_proof(file_path, chunk_index=chunk_index, chunk_size=chunk_size)
```

Ruff (0.15.2)

[error] Local variable `proof` is assigned to but never used

Remove assignment to unused variable `proof`

(F841)